File: tests/basic/list/list.test.ts


import ClassManager from "@base/ClassLoader"
import { createNewTestClasses } from "@base/Stubs"
import TestSystem from "@stub/tests/TestSystem"
import { expect, test } from "bun:test"

test("list manipulation", () => {
	const className = "tests/basic/list/ListTest"
	const stubClasses = createNewTestClasses()

	const classManager = new ClassManager(stubClasses)
	const system = stubClasses.getStubClass("java.lang.System") as TestSystem
	const consoleOutput = system.out

	const main = classManager.get(className)
	main.executeMethod("main", classManager)

  expect(consoleOutput.printlnLines[0]).toBe("0")
  expect(consoleOutput.printlnLines[1]).toBe("[5, 2, 26, 694205]")
  expect(consoleOutput.printlnLines[2]).toBe("[2, 26]")
  expect(consoleOutput.printlnLines[3]).toBe("2")
})

File: tests/basic/list/ListTest.java
package tests.basic.list;

import java.util.ArrayList;
import java.util.List;

public class ListTest {
  
  public static void main(String[] args) {
    
    List<Integer> list = new ArrayList<>();
    System.out.println(list.size());
    list.add(5);
    list.add(2);
    list.add(26);
    list.add(694205);
    System.out.println(list);
    list.remove(3);
    list.remove(0);
    System.out.println(list);
    System.out.println(list.size());

    for(int i : list){
      System.out.println(i);
    }

  }

}
